osf/models/subject.py

```diff
@@ -12,7 +16,9 @@ class Subject(ObjectIDMixin, BaseModel):
     modm_query = None
 
     text = models.CharField(null=False, max_length=256, unique=True)  # max length on prod: 73
-    parents = models.ManyToManyField('self', symmetrical=False, related_name='children')
+    parent = models.ForeignKey('self', related_name='children', null=True, blank=True, on_delete=models.SET_NULL)
+    bepress_subject = models.ForeignKey('self', related_name='aliases', null=True, blank=True)
```

Comment:
Drive-by: `on_delete` should be explicitly set on all ForeignKey fields. It will be a required argument in new versions of Django.